### PR TITLE
added `set_image_index()` to Reader object

### PR DESCRIPTION
**Pull request by MarcinKonowalczyk**
As discussed in #285

### CODE REVIEW

**Comment by MarcinKonowalczyk** [comment]
@almarklein I am not sure how to make the Travis CI build succeed...

File: imageio/core/format.py [hunk 1]
```diff
@@ -350,6 +350,17 @@ def get_next_data(self, **kwargs):
             listed in the documentation of those formats.
             """
             return self.get_data(self._BaseReaderWriter_last_index+1, **kwargs)
+
+        def set_image_index(self, index, **kwargs):
+            """ set_image_index(index)
+
+            Set the pointer to the last read index such that the next call to
+            get_next_data() returns the image specified by the index
```
Comment (almarklein) [inline imageio/core/format.py]: Can you please rephrase this a bit? It contains "internal detail" which will only confuse users. Something like "Set the internal pointer to the given index, such that the next ..." should be enough.

File: tests/test_ffmpeg.py [hunk 2]
```diff
@@ -84,9 +84,20 @@ def test_read_and_write():
...
+        # Set image index
+        R.set_image_index(42)
+        im2 = R.get_next_data()
+        assert im2.shape == (720, 1280, 3)
+
+        R.set_image_index(120)
+        im3 = R.get_next_data()
+        assert im3.shape == (720, 1280, 3)
+        assert (im1 == im3).all()
+        assert not (im1 == im2).all()
```
Comment (almarklein) [inline tests/test_ffmpeg.py]: Thanks, tests!

File: tests/test_ffmpeg.py [hunk 3]
```diff
@@ -140,21 +151,34 @@ def test_reader_more():
...
+
+    # Test loop
     R = imageio.read(get_remote_file('images/realshort.mp4'), 'ffmpeg', loop=1)
     im1 = R.get_next_data()
     for i in range(1, len(R)):
         R.get_next_data()
     im2 = R.get_next_data()
     im3 = R.get_data(0)
-    im4 = R.get_data(2)  # touch skipping frames
+    im4 = R.get_data(2) # touch skipping frames
```
Comment (almarklein) [inline tests/test_ffmpeg.py]: The CI fails over some style issues: https://travis-ci.org/imageio/imageio/jobs/282768272#L558
Flake8 want two spaces before an in-line comment.

**Comment by MarcinKonowalczyk** [comment]
Allright. That should be good. 👍

**Comment by almarklein** [comment]
Awesome, thanks!

**Pull request merged**